### DISCUSSION TITLE
how to release texture or reuse texture in runtime

### DISCUSSION THREAD

**Discussion by dandingol03** [google/filament#4647] [Q&A]
i write a material editor base on the basis of ``sandbox material``, and provide a file dialog to select image texture. I need to reuse the ``Texture`` or release the previous ``Texture``, because different texture size maybe different. If i just call ``setImage``, and provide the specified width and height, can this texture work as as uasal ?

**Comment by bejado** [comment]
A texture's dimensions are fixed once created. The `width` and `height` arguments of `setImage` are for updating a sub-region of the texture image. You'll need to destroy and re-create the Texture if its size changes.

**Reply by dandingol03** [reply]
So i call the ``engine.destroy`` to destroy the texture in runtime ,but the application crash after a seconds. I can't figure out why

**Reply by bejado** [reply]
Make sure the texture isn't being used with any materials before destroying.
